### DISCUSSION TITLE
ESPhttpUpdate bug reports keep getting deleted!?!?

### DISCUSSION THREAD

**Discussion by JimDrewGH** [esp8266/Arduino#8062] [Q&A]
I have posted the same information twice (with the 2nd time even more detailed, although it is not necessary) about the problem with core v3.0.0 has with compiling ANYTHING that uses ESPhttpUpdate in it, yet my reports are being deleted without any notice or explanation.  Almost like this problem is being swept under the rug (which I can't believe is the case).

I have chatted today with several other developers who have the exact same problem when switching to core v3.0.0, so it's not just me and my half a dozen different development machines (on different platforms) that has the issue.  This is a REAL issue with the v3.0.0 core - and there is no mention of this problem ANYWHERE in the v3.0.0 release notes.

### Problem Description

I have used every revision of the ESP8266 core (starting with v2.3.0) with the same project code in numerous commercial applications.  All of these applications use the over-air update capability via ESPhttpUpdate.  When updating to v3.0.0 of the core (through the Arduino IDE's board manager), I find that NONE of my applications will compile due to an error with anything that involves ESPhttpUpdate.  I used the example code found in the core's OTA documentation to base my code on, and those examples also do not compile under v3.0.0 (but do compile with *any* of the previous core versions - I tested them all!)   See the simple and advanced updater section here: **https://arduino-esp8266.readthedocs.io/en/latest/ota_updates/readme.html#http-server**

So, again anything that attempts to use any part of the ESPhttpUpdate class generates an error under the v3.0.0 core.

### Error

_no matching function for call to 'ESP8266HTTPUpdate::update(const char [46])'_

To easily replicate this error, drop this super simple sketch into your Arduino IDE and try to compile it:

```
#include <ESP8266WiFi.h>
#include <ESP8266httpUpdate.h>

void setup() {

  t_httpUpdate_return ret = ESPhttpUpdate.update("http://www.arduino.cc/3p_updates/IoT1172.bin");
}

void loop() {

}
```

If you use the board manager and switch back and forth between core v3.0.0 and any other version you will see that all earlier versions compile just fine, and v3.0.0 generates the same error shown above.  I have tested this on multiple Arduino IDE installations on Windows, Mac, and even Linux (RPi4 mini computer), all with the exact same results.  I did not try different Arduino IDE versions.  I am using v1.8.13 on all of my development setups.

I also tried to use the syntax shown in the documentation example - using a URL, port, and file.  I have been using the single URL method for years now with zero issues.  But for fun, I tried that as well with a similar compiler fail.  Something has apparently changed that has broken the ESPhttpUpdate class.

I have looked all over the v3.0.0 core info and I can't find anything that states that the ESPhttpUpdate class was removed, renamed, etc.  Does anyone know why my **_Issue_** reports would be repeatedly deleted?

**Comment by Pablo2048** [comment]
Both of your "issues" were explained and closed. No one was "deleted" - see https://github.com/esp8266/Arduino/issues?q=is%3Aissue+is%3Aclosed ...

**Comment by d-a-v** [comment]
@JimDrewGH 

ESP8266httpUpdate is relying in ESP8266HttpClient.
It was using a deprecated API that was meant to be removed in future releases.
The release has landed, it is this v3.

This however does not remove ESP8266httpUpdate nor ESP8266HttpClient from the picture.
The API has changed a bit. This change was necessary to make it easier for everybody to use TLS connections without using complex tricks.
Now, ESP8266HttpClient wants a WiFiClient as first argument. So ESP8266httpUpdate needs to give one.
Previously and for quite a long time, the API you were using was throwing a "deprecated" warning.
One should take care of these messages that mean that a new API is present, and the deprecated currently used one is supposed to disappear in a future release.

Your snippet can be updated as [follow](https://github.com/esp8266/Arduino/blob/d7d03f9a41577e6c8ed4437199d73379d83fa5c6/libraries/ESP8266httpUpdate/src/ESP8266httpUpdate.h#L128):

```cpp
void setup() {
  WiFiClient client;
  t_httpUpdate_return ret = ESPhttpUpdate.update(client, "http://www.arduino.cc/3p_updates/IoT1172.bin");
}
```

You can also go and read [the examples](https://github.com/esp8266/Arduino/blob/d7d03f9a41577e6c8ed4437199d73379d83fa5c6/libraries/ESP8266httpUpdate/examples/httpUpdate/httpUpdate.ino#L81) that are supposed to always be compiling and working. There is also an example using TLS in the same directory.

Thank you to make us realise that we forgot to remove dead code linked to the previous API.
Thank you also to point that the documentation has not been updated accordingly, and we're sorry for that. Examples, at least, must always be up to date and the CI system tells us when it is not the case. 
This will be fixed in the 3.0.1 release that will happen very soon.

**Reply by JimDrewGH** [reply]
Thanks for the response from both of you!  I did not realize that there was a "closed" issue area.  I will look at that in the future.

I already have a client initiated, so I will try adding that.

I don't actually get any type of warning for this function using Arduino IDE v1.8.13 and ESP8266 core v2.7.4.  I always run in verbose mode and I definitely do look at anything that pops up as a warning, and I get nothing at all about this.
